packages/shared/lib/services/activity/activity.service.ts

```diff
@@ -265,30 +263,35 @@ export async function getTopLevelLogByEnvironment(
  * each keyed by its associated log ID.
  */
 export async function getLogMessagesForLogs(logIds: number[], environment_id: number): Promise<ActivityLogMessagesGrouped> {
-    if (!logIds.length) {
-        return [];
-    }
-
     try {
+        const limit = 1000;
```

Comment:
I haven't ask before but why did we pick 1000?

Reply:
completely arbitrary during our discussion, but considering the UI is not optimizing the dom it's fair to assume that huge number of lines could potentially crash your browser.
What people do to avoid this is using virtual dom to reuse rows as you scroll instead of just `.map` everything